File: src/main/java/poc/interaction/PersonMutationResolver.java
package poc.interaction;

import com.coxautodev.graphql.tools.GraphQLMutationResolver;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;
import poc.model.Person;
import poc.service.PersonService;

@Component
public class PersonMutationResolver implements GraphQLMutationResolver {
    private final PersonService service;

    @Autowired
    public PersonMutationResolver(PersonService service) {
        this.service = service;
    }

    public Person createPerson(String name, int age) {
        return service.savePerson(new Person(name, age));
    }
}

File: src/main/kotlin/poc/resolver/PersonMutationResolver.kt
package poc.resolver

import com.coxautodev.graphql.tools.GraphQLMutationResolver
import org.dataloader.DataLoader
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Component
import poc.model.Person
import poc.repository.PersonRepository

@Component
class PersonMutationResolver(
    @Autowired private val repository: PersonRepository,
    @Autowired private val personDataLoader: DataLoader<Int, Person>
) : GraphQLMutationResolver {
    fun createPerson(name: String, age: Int) = repository.save(Person(name = name, age = age))
}
